numpyro/infer/mcmc.py

```diff
@@ -382,11 +383,74 @@ def _compile(self, rng_key, *args, extra_fields=(), init_params=None, **kwargs):
         except TypeError:
             pass
 
+    @property
+    def init_state(self):
+        """
+        The initial state of the MCMC chain. If this attribute is None,
+        :meth:`run` will call `self.sampler.init(...)` method for initialization.
+
+        .. warning:: If you want to set a value for this attribute, then you will need to make sure
+            that certain fields such as `HMCState.i` and `HMCState.mean_accept_prob` are
+            reset to 0. In addition, if any of the fields of the initial state have invalid values
+            (e.g. `HMCState.potential_energy` may be invalid if `data` of the new `run` has been
+            changed), the sampler might get stuck or otherwise perform suboptimally.
+            To resolve this issue, you can set `potential_energy` to a large value (say `1e38`),
+            so that the sampler is more likely to accept a new proposal in the first MCMC step
+            (and which will have a correct `potential_energy`). For example
+
+            .. code-block:: python
+
+                mcmc = MCMC(NUTS(model), 100, 100)
+                mcmc.run(random.PRNGKey(0), data0)
+                samples_for_data0 = mcmc.get_samples()
+                mcmc.init_state = mcmc.last_state._replace(
+                    i=0, potential_energy=1e38, mean_accept_prob=0.)
+                mcmc.run(random.PRNGKey(1), data1)
+                samples_for_data1 = mcmc.get_samples()
+
+        """
+        return self._init_state
+
+    @init_state.setter
+    def init_state(self, state):
+        self._init_state = state
+
+    @property
+    def warmup_state(self):
```

Comment:
If we are to expose this, what do you think about calling this `post_warmup_state` for clarity?

Reply:
What do you mean by `post`? Does it mean posterior?

Reply:
I meant post as in `after` as opposed to pre. e.g. `post-warmup draws` from pystan.

Reply:
I have no preference so I'll make the change. Thanks for your suggestion!